File: java/proxy/src/main/CreditManagerProxy.java
package main;

public class CreditManagerProxy extends CreditBase{

    private CreditManager creditManager;
    private double cachedValue = 0;

    @Override
    public double calculate() throws InterruptedException {
        if (creditManager == null)
        {
            creditManager = new CreditManager();
            cachedValue = creditManager.calculate();
        }

        return cachedValue;
    }
}

File: csharp/Proxy/CreditManagerProxy.cs
namespace Proxy;

public class CreditManagerProxy : CalculatorBase
{
    private CreditManager _creditManager;
    private double _cachedValue = 0;

    public override double Calculate()
    {
        if (_creditManager == null)
        {
            _creditManager = new CreditManager();
            _cachedValue = _creditManager.Calculate();
        }

        return _cachedValue;
    }
}